PR: https://github.com/odoo/odoo/pull/194096

From: 5a8be1dfdefaa4876af40302153d9d7c96f175a6
From: Shrey Mehta
Date: 2025-07-19 01:14:26

Breaking data model changes scores: del:2 + add:0, change matches:
-    _inherit = 'onboarding.onboarding.step'
-    payment_onboarding_payment_method = fields.Selection(

Total Changes: 72

[IMP] payment, _*: onboard user with stripe or razorpay

_*= base, sale, {account, website}_payment, payment_{razorpay, stripe}

Prior this commit:
Users were not invited to configure online payments after the removal of the
config bar for onboarding.

Post this commit:
When online payments are activated for Sales or Invoicing, users are onboarded
to configure Stripe Connect or Razorpay directly in the settings.

task-4156274

closes odoo/odoo#194096

Related: odoo/upgrade#7450
Signed-off-by: Antoine Vandevenne (anv) <anv@odoo.com>

================================= pseudo patch: =================================

--- a/addons/payment/models/__init__.py
+++ b/addons/payment/models/__init__.py
@@ -1,7 +1,6 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import ir_http
-from . import onboarding_step
 from . import payment_method
 from . import payment_provider
 from . import payment_token

--- a/addons/payment/models/onboarding_step.py
+++ b/None
@@ -1,12 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import api, models
-
-
-class OnboardingOnboardingStep(models.Model):
-    _inherit = 'onboarding.onboarding.step'
-
-    @api.model
-    def action_validate_step_payment_provider(self):
-        """ Override of `onboarding` to validate other steps as well. """
-        return self.action_validate_step('payment.onboarding_onboarding_step_payment_provider')

--- a/addons/payment/models/payment_provider.py
+++ b/addons/payment/models/payment_provider.py
@@ -443,6 +443,18 @@ class PaymentProvider(models.Model):
                 'tag': 'reload',
             }
 
+    def action_start_onboarding(self, menu_id=None):
+        """Start the provider-specific onboarding.
+
+        Providers implementing a specific onboarding must override this method and return the action
+        to run the onboarding.
+
+        :param int menu_id: The menu from which the onboarding is started, as an `ir.ui.menu` id.
+        :return: The onboarding action.
+        :rtype: dict
+        """
+        return {}
+
     def action_toggle_is_published(self):
         """ Toggle the field `is_published`.
 

--- a/addons/payment/models/res_company.py
+++ b/addons/payment/models/res_company.py
@@ -1,20 +1,11 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, fields, models
+from odoo import api, models
 
 
 class ResCompany(models.Model):
     _inherit = 'res.company'
 
-    payment_onboarding_payment_method = fields.Selection(
-        string="Selected onboarding payment method",
-        selection=[
-            ('paypal', "PayPal"),
-            ('stripe', "Stripe"),
-            ('manual', "Manual"),
-            ('other', "Other"),
-        ])
-
     @api.model_create_multi
     def create(self, vals_list):
         companies = super().create(vals_list)
@@ -31,39 +22,3 @@ class ResCompany(models.Model):
                 provider_sudo.copy({'company_id': company.id})
 
         return companies
-
-    def _run_payment_onboarding_step(self, menu_id=None):
-        """ Install the suggested payment modules and configure the providers.
-
-        It's checked that the current company has a Chart of Account.
-
-        :param int menu_id: The menu from which the user started the onboarding step, as an
-                            `ir.ui.menu` id
-        :return: The action returned by `action_stripe_connect_account`
-        :rtype: dict
-        """
-        self.env.company.get_chart_of_accounts_or_fail()
-
-        self._install_modules(['payment_stripe'])
-
-        # Create a new env including the freshly installed module(s)
-        new_env = api.Environment(self.env.cr, self.env.uid, self.env.context)
-
-        # Configure Stripe
-        stripe_provider = new_env['payment.provider'].search([
-            *self.env['payment.provider']._check_company_domain(self.env.company),
-            ('code', '=', 'stripe')
-        ], limit=1)
-        if not stripe_provider:
-            base_provider = self.env.ref('payment.payment_provider_stripe')
-            # Use sudo to access payment provider record that can be in different company.
-            stripe_provider = base_provider.sudo().with_context(
-                stripe_connect_onboarding=True,
-            ).copy(default={'company_id': self.env.company.id})
-
-        return stripe_provider.action_stripe_connect_account(menu_id=menu_id)
-
-    def _install_modules(self, module_names):
-        modules_sudo = self.env['ir.module.module'].sudo().search([('name', 'in', module_names)])
-        STATES = ['installed', 'to install', 'to upgrade']
-        modules_sudo.filtered(lambda m: m.state not in STATES).button_immediate_install()
